pkg/operator/ceph/object/health.go

```diff
@@ -119,37 +125,52 @@ func (c *bucketChecker) checkObjectStoreHealth() error {
 
 		Always keep the bucket and the user for the health check, just do PUT and GET because bucket creation is expensive
 	*/
+	var tlsCert []byte
 	var s3AccessKey string
 	var s3SecretKey string
-	var tlsCert []byte
+	var err error
 	s3endpoint := buildDNSEndpoint(BuildDomainName(c.objContext.Name, c.namespacedName.Namespace), c.port, c.objectStoreSpec.IsTLSEnabled())
 
 	// Generate unique user and bucket name
 	bucketName := genUniqueBucketName(c.objContext.UID)
 	userConfig := c.genUserConfig()
 
-	// Create S3 user
-	logger.Debugf("creating s3 user object %q for object store %q", userConfig.UserID, c.namespacedName.Name)
-	user, rgwerr, err := CreateUser(c.objContext, userConfig)
+	var httpClient *http.Client
+	if c.objectStoreSpec.IsTLSEnabled() {
+		tlsCert, err = GetTlsCaCert(c.objContext, c.objectStoreSpec)
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch CA cert for the user %q to establish TLS connection with the object store %q", userConfig.ID, c.namespacedName.Name)
+		}
+		httpClient.Transport = BuildTransportTLS(tlsCert)
+	}
+
+	// Build admin Ops API connection
+	co, err := admin.New(s3endpoint, c.adminOpsUserAccessKey, c.adminOpsUserSecretKey, httpClient)
+	if err != nil {
+		return errors.Wrap(err, "failed to build admin ops API connection")
+	}
+	c.adminOpsClient = co
+
+	// Create checker user
+	logger.Debugf("creating s3 user object %q for object store %q", userConfig.ID, c.namespacedName.Name)
+	user, err := co.CreateUser(context.Background(), userConfig)
 	if err != nil {
-		if rgwerr == ErrorCodeFileExists {
-			user, _, err = GetUser(c.objContext, userConfig.UserID)
+		if errors.Is(err, admin.ErrUserExists) {
+			objectUser, err := co.GetUser(context.Background(), userConfig)
 			if err != nil {
-				return errors.Wrapf(err, "failed to get details from ceph object user %q for object store %q", userConfig.UserID, c.namespacedName.Name)
+				return errors.Wrapf(err, "failed to get details from ceph object user %q", userConfig.ID)
 			}
-		} else {
-			return errors.Wrapf(err, "failed to create object user %q. error code %d for object store %q", userConfig.UserID, rgwerr, c.namespacedName.Name)
-		}
-	}
-	// Set access and secret key
-	s3AccessKey = *user.AccessKey
-	s3SecretKey = *user.SecretKey
 
-	if c.objectStoreSpec.IsTLSEnabled() {
-		tlsCert, err = GetTlsCaCert(c.objContext, c.objectStoreSpec)
-		if err != nil {
-			return errors.Wrapf(err, "failed to fetch CA cert for the user %q to establish TLS connection with the object store %q", userConfig.UserID, c.namespacedName.Name)
+			// Set access and secret key
+			s3AccessKey = objectUser.Keys[0].AccessKey
+			s3SecretKey = objectUser.Keys[0].SecretKey
+		} else {
+			return errors.Wrapf(err, "failed to create ceph object user %q", userConfig.ID)
 		}
+	} else {
```

Comment:
Will all error cases return before this? Seems like this else block isn't needed, and also wouldn't need lines 165-166 that seem duplicated above.
```suggestion
	} 
	// Set access and secret key
	s3AccessKey = user.Keys[0].AccessKey
	s3SecretKey = user.Keys[0].SecretKey<br class="Apple-interchange-newline">
```